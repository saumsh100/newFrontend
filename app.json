  {
  "name": "carecru",
  "scripts": {
  },
  "env": {
    "NODE_ENV": {
      "required": true
    },
    "RETHINKDB_AUTHKEY": {
      "required": true
    },
    "RETHINKDB_DB": {
      "required": true
    },
    "RETHINKDB_HOST": {
      "required": true
    },
    "RETHINKDB_PORT": {
      "required": true
    },
    "VENDASTA_API_KEY": {
      "required": true
    },
    "VENDASTA_API_USER": {
      "required": true
    },
    "TWILIO_ACCOUNT_SID": {
      "required": true
    },
    "TWILIO_AUTH_TOKEN": {
      "required": true
    },
    "TWILIO_NUMBER": {
      "required": true
    }
  },
  "formation": {
    "web": {
      "quantity": 1
    }
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
