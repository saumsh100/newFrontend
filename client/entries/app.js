
import '../apps/default';
import '../apps/Dashboard';
