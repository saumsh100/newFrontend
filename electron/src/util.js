
module.exports.getChannelFromVersion = (version) => {
  const splitVersion = version.split('-');
  return splitVersion[1] || null;
};
