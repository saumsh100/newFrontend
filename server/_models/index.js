
const Sequelize = require('sequelize');
const { postgres } = require('../config/globals');

const sequelizeConfig = {
  logging: false,
  dialect: 'postgres',
  host: postgres.host,
  port: postgres.port,
  define: {
    underscored: false,
    paranoid: true,
  },
};

// Don't use NODE_ENV===production
// It doesn't let us debug production locally easily
if (postgres.ssl) {
  sequelizeConfig.dialectOptions = {
    ssl: true,
  };
}

// If true, sequelize will dump all PostgreSQL queries into terminal
if (postgres.logging) {
  sequelizeConfig.logging = console.log // eslint-disable-line
}

// initialize sequelize
const sequelize = new Sequelize(
  postgres.database,
  postgres.username,
  postgres.password,
  sequelizeConfig,
  );

// Test the connection for logs
sequelize
  .authenticate()
  .then(() => {
    console.log(`Sequelize connected to database=${postgres.database} at host:port=${postgres.host}:${postgres.port}`);
  })
  .catch(err => {
    console.error('postgres config', postgres);
    console.error('Unable to connect to the database:', err);
  });

// Import and store all models.
const models = [];
// models.push((require('./Address').default(sequelize, Sequelize)));
models.push((require('./Account').default(sequelize, Sequelize)));
models.push((require('./Appointment').default(sequelize, Sequelize)));
models.push((require('./AuthSession').default(sequelize, Sequelize)));
models.push((require('./Call').default(sequelize, Sequelize)));
models.push((require('./Chair').default(sequelize, Sequelize)));
models.push((require('./Chat').default(sequelize, Sequelize)));
models.push((require('./Enterprise').default(sequelize, Sequelize)));
models.push((require('./Family').default(sequelize, Sequelize)));
models.push((require('./Invite').default(sequelize, Sequelize)));
models.push((require('./Patient').default(sequelize, Sequelize)));
models.push((require('./PatientUser').default(sequelize, Sequelize)));
models.push((require('./Permission').default(sequelize, Sequelize)));
models.push((require('./PinCode').default(sequelize, Sequelize)));
models.push((require('./Practitioner').default(sequelize, Sequelize)));
models.push((require('./Practitioner_Service').default(sequelize, Sequelize)));
models.push((require('./PractitionerRecurringTimeOff').default(sequelize, Sequelize)));
models.push((require('./Recall').default(sequelize, Sequelize)));
models.push((require('./Reminder').default(sequelize, Sequelize)));
models.push((require('./Request').default(sequelize, Sequelize)));
models.push((require('./Segment').default(sequelize, Sequelize)));
models.push((require('./SentRecall').default(sequelize, Sequelize)));
models.push((require('./SentReminder').default(sequelize, Sequelize)));
models.push((require('./Service').default(sequelize, Sequelize)));
models.push((require('./SyncClientError').default(sequelize, Sequelize)));
models.push((require('./SyncClientVersion').default(sequelize, Sequelize)));
models.push((require('./TextMessage').default(sequelize, Sequelize)));
models.push((require('./Token').default(sequelize, Sequelize)));
models.push((require('./User').default(sequelize, Sequelize)));
models.push((require('./WaitSpot').default(sequelize, Sequelize)));
models.push((require('./WeeklySchedule').default(sequelize, Sequelize)));

const db = {};
models.forEach((model) => {
  db[model.name] = model;
});

Object.keys(db).forEach((modelName) => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
