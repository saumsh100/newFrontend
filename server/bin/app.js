
require('../config/initializeCodeTranspiler');

const globals = require('../config/globals');
const applyWebpack = require('../config/webpack/applyWebpack');
const handleErrors = require('../middleware/handleErrors');
const chalk = require('chalk');
const app = require('../config/express');

// require('../config/kue');

// Uses the NODE_ENV to determine logging full stack or not
app.set('showStackError', true);

// Load in webpack configurations
applyWebpack(app);


// Set the Handlebars templating engine
app.set('views', `${globals.root}/views`);
app.engine('hbs', require('../config/handlebars').engine);
app.set('view engine', 'hbs');

// Extra logging for communication with server
app.use((req, res, next) => {
  console.log(chalk.blue(req.method, req.originalUrl));
  next();
});


// Bind the routes

app.use(require('../routes'));

// Catch errors, log and respond to client
app.use(handleErrors);

// assume 404 since no middleware responded
/*app.use((req, res) => {
  res.status(404).render('404', {
    url: req.originalUrl,
    error: 'Not found',
  });
});*/

module.exports = app;
