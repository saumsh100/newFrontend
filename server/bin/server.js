
const http = require('http');
const globals = require('../config/globals');
const app = require('./app');
const Availability = require('../models/Availability');
const Patient = require('../models/Patient');

const server = http.createServer(app);

const io = require('socket.io')(server);

// TODO: implement Thinky!!!
io.on('connection', (socket) => {
  console.log('a new socket connected');
  socket.on('fetchAvailabilities', () => {
    Availability.run()
      .then((availabilities) => {
        socket.emit('receiveAvailabilities', availabilities);
      });
  });
  
  socket.on('fetchPatients', () => {
    Patient.run()
      .then((patients) => {
        socket.emit('receivePatients', patients);
      });
  });
  
  socket.on('addAvailability', (data) => {
    new Availability(data).save().then((availability) => {
      io.sockets.emit('availabilityAdded', availability);
    });
  });
  
  socket.on('removeAvailability', (data) => {
    Availability.get(data.id).then((availability) => {
      availability.delete()
        .then((result) => {
          io.sockets.emit('availabilityRemoved', result);
        });
    });
  });
  
  socket.on('syncClientTrigger', (data) => {
    console.log('sync client triggers!', 'data:', data);
  });
});

server.listen(globals.port, () => {
  console.log(`CareCru HTTP Server is running on port ${globals.port}`);
});
