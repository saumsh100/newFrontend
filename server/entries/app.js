
import '../../client/apps/default';
import '../../client/apps/Dashboard';
