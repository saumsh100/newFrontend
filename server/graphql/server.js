
import { Router } from 'express';
import graphQLHTTP from 'express-graphql';
import { sequelizeAuthMiddleware } from '../middleware/auth';
import schema from './data/schema';

const GraphQLRouter = Router();

const PORT = 4000;

const GraphQLServer = graphQLHTTP({
  schema,
  graphiql: true,
  pretty: true,
  subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`,
});

// Reusing auth middleware as the server is exposed the same way an API would.
GraphQLRouter.all('*', sequelizeAuthMiddleware, GraphQLServer);

export default GraphQLRouter;

