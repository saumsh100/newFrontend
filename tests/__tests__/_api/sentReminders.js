
import request from 'supertest';
import app from '../../../server/bin/app';
import generateToken from '../../_util/generateToken';
import { SentReminder, Reminder, Appointment, Patient } from '../../../server/_models';
import wipeModel, { wipeAllModels } from '../../_util/wipeModel';
import { accountId, seedTestUsers, wipeTestUsers } from '../../_util/seedTestUsers';
import { patientId } from '../../_util/seedTestPatients';
import { reminderId1, seedTestReminders } from '../../_util/seedTestReminders';
import { appointmentId, seedTestAppointments } from '../../_util/seedTestAppointments';
import { omitPropertiesFromBody } from '../../util/selectors';

const rootUrl = '/_api/sentReminders';

const sentReminderId = 'e757afb0-14ef-4763-b162-c573169131c1';
const sentReminder = {
  id: sentReminderId,
  reminderId: reminderId1,
  accountId,
  patientId,
  appointmentId,
  isSent: true,
  lengthSeconds: 540,
  primaryType: 'email',
  createdAt: '2017-07-19T00:14:30.932Z',
};

async function seedTestSentReminder() {
  await seedTestAppointments();
  await seedTestReminders();
  await SentReminder.create(sentReminder);
}

describe('/api/sentReminders', () => {
  // Seed with some standard user data
  let token = null;
  beforeEach(async () => {
    await wipeModel(SentReminder);
    await seedTestUsers();
    await seedTestSentReminder();
    token = await generateToken({ username: 'manager@test.com', password: '!@CityOfBudaTest#$' });
  });


  afterAll(async () => {
    await wipeModel(SentReminder);
    await wipeModel(Reminder);
    await wipeModel(Appointment);
    await wipeModel(Patient);
    await wipeTestUsers();
  });

  describe('GET /', () => {
    test('retrieve sent reminder', () => {
      return request(app)
        .get(`${rootUrl}?join=appointment,reminder,patient`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body, ['homePhoneNumber', 'insurance', 'otherPhoneNumber', 'prefPhoneNumber', 'workPhoneNumber']);

          expect(body).toMatchSnapshot();
        });
    });
  });
});
