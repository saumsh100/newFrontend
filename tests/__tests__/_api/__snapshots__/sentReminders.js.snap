// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/sentReminders GET / retrieve sent reminder 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isReminderSent": true,
        "isShortCancelled": false,
        "isSyncedWithPms": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
    "patients": Object {
      "10518e11-b9d2-4d74-9887-29eaae7b5938": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "address": null,
        "birthDate": "1980-01-01T00:00:00.932Z",
        "email": "testpatient@test.com",
        "familyId": null,
        "firstName": "Ronald",
        "gender": null,
        "id": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "isDeleted": false,
        "isSyncedWithPms": false,
        "language": null,
        "lastName": "Mcdonald",
        "middleName": null,
        "mobilePhoneNumber": "+17789999999",
        "patientUserId": null,
        "phoneNumber": null,
        "pmsCreatedAt": null,
        "pmsId": "12",
        "prefContactPhone": null,
        "prefName": null,
        "preferences": Object {
          "afternoon": true,
          "birthdayMessage": true,
          "emailNotifications": true,
          "evening": true,
          "morning": true,
          "newsletter": true,
          "phone": true,
          "reminders": true,
          "sms": true,
          "weekdays": true,
          "weekends": true,
        },
        "status": "Active",
        "type": null,
      },
    },
    "reminders": Object {
      "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "id": "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
        "lengthSeconds": null,
        "primaryType": "sms",
      },
    },
    "sentReminders": Object {
      "e757afb0-14ef-4763-b162-c573169131c1": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "appointment": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "appointmentId": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "id": "e757afb0-14ef-4763-b162-c573169131c1",
        "isConfirmable": true,
        "isConfirmed": false,
        "isSent": true,
        "lengthSeconds": 540,
        "patient": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "primaryType": "email",
        "reminder": "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
        "reminderId": "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
      },
    },
  },
  "result": Array [
    "e757afb0-14ef-4763-b162-c573169131c1",
  ],
}
`;
